### PR TITLE
Optimize Kafka queue offset handling

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

A way to optimize offsets and to avoid weird situations.

## Checklist

-   [ ] Jest tests

### CODE REVIEW

File: src/ingestion/kafka-queue.ts [hunk 1]
```diff
@@ -1,13 +1,48 @@
 import * as Sentry from '@sentry/node'
 import { Kafka, Consumer, Message, EachBatchPayload } from 'kafkajs'
 import { PluginsServer, Queue, RawEventMessage } from 'types'
-import { KAFKA_EVENTS_INGESTION_HANDOFF } from './topics'
+import { KAFKA_EVENTS_WAL } from './topics'
```
Comment (mariusandra) [inline src/ingestion/kafka-queue.ts:4]: Why the change from `KAFKA_EVENTS_INGESTION_HANDOFF`?

**Review by mariusandra — COMMENTED** [review]
I feel like this code is way too complex for what it should do. It also leaves the kafka_offset inside event.

I tried another way to get the same result with https://github.com/PostHog/plugin-server/pull/124 ... though I haven't run the code yet.

**Comment by Twixes** [comment]
Oh, sry, actually I was doing some debugging due to tests failing, so this is not in a reviewable state ATM (was reviewable at 6feda22).

If this is too complex, what would your alternative suggestion be? This isn't super simple, but would cover all cases pretty well (and there are various ones we have to consider, e.g. some plugin: added events there never were in the Kafka queue, or removed a series of events that were in the Kafka queue, or removed all of them, or removed every other event, or removed all of them).

**Comment by mariusandra** [comment]
Which cases would #124 not cover?

**Comment by mariusandra** [comment]
just saw your reply :), will give it a thought after dinner

File: src/ingestion/kafka-queue.ts [hunk 2]
```diff
@@ -71,7 +102,7 @@ export class KafkaQueue implements Queue {
             }
             const singleIngestionTimer = new Date()
             await this.saveEvent(event)
-            resolveOffset(event.kafka_offset!)
+            resolveOffset(eventUuidToKafkaOffset.get(event.uuid!)!)
```
Comment (Twixes) [inline src/ingestion/kafka-queue.ts:105]: TODO: don't run this if the event was not present in batch from Kafka (meaning it was inserted by a plugin's `processEventBatch`)
